File: Second Semester(Java and C++)/List 3/Java/Task 1/Rectangle.java
import java.util.Comparator;

public class Rectangle extends Quadrangle {
    public Rectangle(int a, int b, int c, int d, int angle) throws IllegalArgumentException {
        sides.add(a);
        sides.add(b);
        sides.add(c);
        sides.add(d);
        this.angle = angle;

        sides.sort(Comparator.naturalOrder());

        if ((sides.get(0) != sides.get(1)) || (sides.get(2) != sides.get(3)) || (angle != 90)) {
            throw new IllegalArgumentException("This is not a rectangle!");
        }
    }

    @Override
    public String getName() {
        return "Rectangle";
    }
}

File: Second Semester(Java and C++)/List 3/C++/Rectangle.cpp
#include "Rectangle.hpp"

#include <stdexcept>
#include <algorithm>

Rectangle::Rectangle(int a, int b, int c, int d, int angle) noexcept(false) {
    sides.push_back(a);
    sides.push_back(b);
    sides.push_back(c);
    sides.push_back(d);
    this->angle = angle;

    sort(sides.begin(), sides.end());

    if ((sides[0] != sides[1]) || (sides[2] != sides[3]) || (angle != 90)) {
        throw std::invalid_argument("This is not a rectangle(invalid sides or angle)");
    }
}

std::string Rectangle::getName() noexcept(true) {
    return "Rectangle";
}